extensions/jackson/runtime/src/main/java/io/quarkus/jackson/runtime/JacksonBuildTimeConfig.java

```diff
@@ -4,74 +4,72 @@
 import java.util.Optional;
 
 import com.fasterxml.jackson.annotation.JsonInclude;
-import com.fasterxml.jackson.databind.PropertyNamingStrategies;
-
-import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
 
 @ConfigRoot(phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
-public class JacksonBuildTimeConfig {
+@ConfigMapping(prefix = "quarkus.jackson")
+public interface JacksonBuildTimeConfig {
 
     /**
      * If enabled, Jackson will fail when encountering unknown properties.
      * <p>
      * You can still override it locally with {@code @JsonIgnoreProperties(ignoreUnknown = false)}.
      */
-    @ConfigItem(defaultValue = "false")
-    public boolean failOnUnknownProperties;
+    @WithDefault("false")
+    Optional<Boolean> failOnUnknownProperties();
```

Comment:
The type shouldn't be changed here. We have a default value so it shouldn't be `Optional`.

Same for all the others you changed.